runtime/compiler/z/codegen/J9CodeGenerator.cpp

```diff
@@ -106,6 +106,13 @@ J9::Z::CodeGenerator::initialize()
       cg->setSupportsInlineStringLatin1Inflate();
       }
 
+   static bool disableInlineStringCodingHasNegatives = feGetEnv("TR_DisableInlineStringCodingHasNegatives") != NULL;
+   if (cg->getSupportsVectorRegisters() && !disableInlineStringCodingHasNegatives &&
+        !TR::Compiler->om.canGenerateArraylets())
```

Comment:
Hmm, should we open up issue to denote that we need to implement / check this for off-heap?